esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -439,47 +439,78 @@ void WUDetails::processRequest(IEspWUDetailsRequest &req, IEspWUDetailsResponse
+        Owned<IConstWUScopeIterator> iter = &workunit->getScopeIterator(wuScopeFilter);
+        ForEach(*iter)
+        {
+            StatisticScopeType scopeType = iter->getScopeType();
+            const char * scope = iter->queryScope();
+            assertex(scope);
 
-        wuDetailsVisitor.noteScopeType(scopeType);
-        iter->playProperties(wuDetailsVisitor);
+            wuDetailsVisitor.noteScopeType(scopeType);
+            iter->playProperties(wuDetailsVisitor);
 
-        Owned<IEspWUResponseScope> respScope = createWUResponseScope("","");
-        if (includeScope)
-            respScope->setScopeName(scope);
-        if (includeScopeType)
-            respScope->setScopeType(queryScopeTypeName(scopeType));
-        if (includeId)
-            respScope->setId(queryScopeTail(scope));
+            Owned<IEspWUResponseScope> respScope = createWUResponseScope("","");
+            if (includeScope)
+                respScope->setScopeName(scope);
+            if (includeScopeType)
+                respScope->setScopeType(queryScopeTypeName(scopeType));
+            if (includeId)
+                respScope->setId(queryScopeTail(scope));
 
-        IArrayOf<IEspWUResponseProperty> & properties = wuDetailsVisitor.getResponseProperties();
-        if (!properties.empty())
-            respScope->setProperties(properties);
+            IArrayOf<IEspWUResponseProperty> & properties = wuDetailsVisitor.getResponseProperties();
+            if (!properties.empty())
+                respScope->setProperties(properties);
 
-        IArrayOf<IEspWUResponseNote> & notes = wuDetailsVisitor.getResponseNotes();
-        if (!notes.empty())
-            respScope->setNotes(notes);
+            IArrayOf<IEspWUResponseNote> & notes = wuDetailsVisitor.getResponseNotes();
+            if (!notes.empty())
+                respScope->setNotes(notes);
 
-        respScopes.append(*respScope.getClear());
+            respScopes.append(*respScope.getClear());
 
-        wuDetailsVisitor.resetScope();
+            wuDetailsVisitor.resetScope();
+        }
+        maxVersion.append(wuDetailsVisitor.getMaxTimestamp());
     }
-    StringBuffer maxVersion;
-    maxVersion.append(wuDetailsVisitor.getMaxTimestamp());
+    else
+    {
+        if (req.getPropertiesToReturn().getAllNotes())
+        {
+            Linked<IPropertyTree> componentConfig = getComponentConfig();
+            IArrayOf<IEspWUResponseNote> espWuResponseNotes;
 
+            Owned<IPropertyTreeIterator> iter = componentConfig->getElements("warnings");
+            ForEach(*iter)
+            {
+                IPropertyTree & cur = iter->query();
+
+                Owned<IEspWUResponseNote> espWuResponseNote = createWUResponseNote("","");
+                espWuResponseNote->setSource(cur.queryProp("@source"));
+                espWuResponseNote->setMessage(cur.queryProp("@msg"));
+                if (cur.queryProp("@code"))
```

Comment:
trivial:  cur.hasProp() is the correct function to test for existence (although query will work)